crates/core/tedge_agent/src/entity_manager/server.rs

```diff
@@ -31,11 +36,40 @@ pub enum EntityStoreRequest {
 pub enum EntityStoreResponse {
     Get(Option<EntityMetadata>),
     Create(Result<Vec<RegisteredEntityData>, entity_store::Error>),
+    Patch(Result<(), entity_store::Error>),
     Delete(Vec<EntityMetadata>),
     List(Vec<EntityMetadata>),
     Ok,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct EntityTwinData {
+    pub topic_id: EntityTopicId,
+    #[serde(flatten)]
+    pub fragments: Map<String, Value>,
+}
+
+impl EntityTwinData {
+    pub fn try_new(
+        topic_id: EntityTopicId,
+        twin_data: Map<String, Value>,
+    ) -> Result<Self, InvalidTwinData> {
+        for key in twin_data.keys() {
+            if key.starts_with('@') {
+                return Err(InvalidTwinData);
+            }
+        }
+        Ok(Self {
+            topic_id,
+            fragments: twin_data,
+        })
+    }
+}
+
+#[derive(thiserror::Error, Debug, Clone, PartialEq, Eq)]
+#[error("Fragment keys starting with '@' are not allowed as twin data")]
+pub struct InvalidTwinData;
```

Comment:
It would be good to tell which key has been rejected.

Reply:
Resolved by 6e7c4c6